src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/DfsServiceTests.cs

```diff
@@ -0,0 +1,305 @@
+            ExceptionAssert.Throws<Exception>(() => _dfs1.StartAsync().Wait());
+            await _dfs1.StopAsync();
+            Assert.Equal(0, peer.Addresses.Count());
+        }
+
+        [Fact]
+        public async Task Can_Start_And_Stop_MultipleEngines()
+        {
+            var peer1 = _dfs1.LocalPeer;
+            var peer2 = _dfs2.LocalPeer;
+
+            for (int n = 0; n < 3; ++n)
+            {
+                await _dfs1.StartAsync();
+                Assert.NotEqual(0, peer1.Addresses.Count());
+                await _dfs2.StartAsync();
+                Assert.NotEqual(0, peer2.Addresses.Count());
+
+                await _dfs2.StopAsync();
+                Assert.Equal(0, peer2.Addresses.Count());
+                await _dfs1.StopAsync();
+                Assert.Equal(0, peer1.Addresses.Count());
+            }
+        }
+
+        [Fact]
+        public async Task Can_Use_Private_Node()
+        {
+            using (var ipfs = TestDfs.GetTestDfs(_output))
+            {
+                ipfs.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+                ipfs.Options.Swarm.PrivateNetworkKey = new PreSharedKey().Generate();
+                await _dfs1.StartAsync();
+            }
+        }
+
+        [Fact]
+        public async Task LocalPeer()
+        {
+            Task<Peer>[] tasks =
+            {
+                Task.Run(() => _dfs1.LocalPeer),
+                Task.Run(() => _dfs1.LocalPeer)
+            };
+            var r = await Task.WhenAll(tasks);
+            Assert.Equal(r[0], r[1]);
+        }
+
+        [Fact]
+        public async Task KeyChain()
+        {
+            Task<IKeyApi>[] tasks =
+            {
+                Task.Run(() => _dfs1.KeyApi),
+                Task.Run(() => _dfs1.KeyApi)
+            };
+            var r = await Task.WhenAll(tasks);
+            Assert.Equal(r[0], r[1]);
+        }
+
+        [Fact]
+        public async Task KeyChain_GetKey()
+        {
+            var keyChain = await _dfs1.KeyChainAsync();
+            var key = await keyChain.GetPrivateKeyAsync("self");
+            Assert.NotNull(key);
+            Assert.True(key.IsPrivate);
+        }
+
+        [Fact]
+        public async Task Swarm_Gets_Bootstrap_Peers()
+        {
+            var bootPeers = (await _dfs1.BootstrapApi.ListAsync()).ToArray();
+            await _dfs1.StartAsync();
+            try
+            {
+                var swarm = _dfs1.SwarmService;
+                var knownPeers = swarm.KnownPeerAddresses.ToArray();
+                var endTime = DateTime.Now.AddSeconds(3);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                    {
+                        throw new XunitException("Bootstrap peers are not known.");
+                    }
+
+                    if (bootPeers.All(a => knownPeers.Contains(a)))
+                    {
+                        break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)